pkg/jobs/utils/helpers.go

```diff
@@ -465,3 +496,40 @@ func parseVersionInfo(msg string) (channel, upstream string, semversion semver.V
 
 	return channel, upstream, semversion, nil
 }
+
+func GetClusterType(
+	hiveset hiveclient.Interface,
+	dc dynamic.Interface,
+	clusterName string,
+	clusterNamespace string) (string, error) {
+	// shortcut for Hypershift cluster
+	if clusterName != clusterNamespace {
+		return HypershiftClusterType, nil
+	}
+
+	// if clusterName and clusterNamespace are equal we need more info
+	cluster, err := hiveset.HiveV1().ClusterDeployments(clusterName).Get(context.TODO(), clusterName, v1.GetOptions{})
+	if err == nil && cluster != nil {
+		return StandaloneClusterType, nil
+	}
```

Comment:
Might need to do if it's not a notfounderror then return the err. So something like:
```suggestion
	} else if !apierrors.isnotfound(err) {
	  return "", err
	}
```